### PR TITLE
CLOUDP-65697: Add init container and volumes to architecture doc

### PR DESCRIPTION
**Pull request by fabianlindfors**
### All Submissions:

* [ ] Have you opened an Issue before filing this PR?
* [X] Have you signed our [CLA](https://www.mongodb.com/legal/contributor-agreement)?
* [X] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?
* [ ] Put `closes #XXXX` in your comment to auto-close the issue that your PR fixes (if such).

This PR adds a description of the pre-stop hook init container and the volumes to `architecture.md`. There was a TODO comment for this.

### CODE REVIEW

File: architecture.md [hunk 1]
```diff
@@ -13,14 +13,24 @@ The MongoDB Community Kubernetes Operator is a [Custom Resource Definition](http
 You create and update MongoDB resources by defining a MongoDB resource definition. When you apply the MongoDB resource definition to your Kubernetes environment, the Operator:
 
 1. Creates a [StatefulSet](https://kubernetes.io/docs/concepts/workloads/controllers/statefulset/) that contains one [pod](https://kubernetes.io/docs/concepts/workloads/pods/pod-overview/) for each [replica set](https://docs.mongodb.com/manual/replication/) member. 
-1. Creates two [containers](https://kubernetes.io/docs/concepts/containers/overview/) in each pod:
+1. Writes the Automation configuration as a [ConfigMap](https://kubernetes.io/docs/concepts/configuration/configmap/) and mounts it to each pod. 
+1. Creates one [init container](https://kubernetes.io/docs/concepts/workloads/pods/init-containers/) and two [containers](https://kubernetes.io/docs/concepts/containers/overview/) in each pod:
+
+   - An init container which copies the `cmd/prestop` binary to the main `mongod` container. </br>
```
Comment (rodrigovalin) [inline architecture.md:19]: what's the `</br>` doing here?

File: architecture.md [hunk 2]
```diff
@@ -13,14 +13,24 @@ The MongoDB Community Kubernetes Operator is a [Custom Resource Definition](http
...
      `mongod` is the primary daemon process for the MongoDB system. It handles data requests, manages data access, and performs background management operations.
 
    - A container for the MongoDB Agent. </br>
      The Automation function of the MongoDB Agent handles configuring, stopping, and restarting the `mongod` process. The MongoDB Agent periodically polls the `mongod` to determine status and can deploy changes as needed. 
-1. Writes the Automation configuration as a [ConfigMap](https://kubernetes.io/docs/concepts/configuration/configmap/) and mounts it to each pod. 
+     
+1. Creates several volumes:
+
+   - `data-volume` which is [persistent](https://kubernetes.io/docs/concepts/storage/persistent-volumes/) and mounts to `/data` on both the server and agent containers. Stores server data as well as `automation-mongod.conf` written by the agent and some locks the agent needs.
+   - `automation-config` which holds the mounted Automation config used by both the server and agent.
```
Comment (rodrigovalin) [inline architecture.md:31]: maybe indicate that this is ephemeral storage (mounted from a configmap)?

File: architecture.md [hunk 3]
```diff
@@ -13,14 +13,24 @@ The MongoDB Community Kubernetes Operator is a [Custom Resource Definition](http
...
+   - `healthstatus` which contains the agent's current status. This is shared with the `mongod` container where it's used by the pre-stop hook.
```
Comment (rodrigovalin) [inline architecture.md:32]: this is also ephemeral and lasts for as long as the Pod where it resides.

File: architecture.md (hunk 1 above)
Comment (fabianlindfors) [reply architecture.md:19]: There's a break for every item in the containers list since before. Should we remove that?

Comment (rodrigovalin) [reply architecture.md:19]: Can you check if it is necessary? if not needed, please remove 👍

Comment (fabianlindfors) [reply architecture.md]: I think it's fine without them, removed.

**Pull request merged**